bin/upload-to-s3

```diff
@@ -34,6 +34,8 @@ main() {
         echo "Uploading $src → $dst"
         if [[ "$dst" == *.gz ]]; then
             gzip -c "$src"
+        elif  [[ "$dst" == *.xz ]]; then
+            xz -2 -c "$src"
         else
             cat "$src"
         fi | aws s3 cp --no-progress - "$dst" --metadata sha256sum="$src_hash" "$(content-type "$dst")" "$(content-encoding "$dst")"
```

Comment:
I believe the GitHub action is failing at the this `aws s3 cp` step. 
Looks like `content-type` and `content-encoding` also need to be updated to expect `.xz` files.

Reply:
I attempted to address this in 3ef9082ee9f1070c974bbf9e1c5c655f09d1d804

https://github.com/nextstrain/ncov-ingest/blob/3ef9082ee9f1070c974bbf9e1c5c655f09d1d804/bin/upload-to-s3#L56-L74

Considerations:
 - Do we even need to set these in the first place?
 - The content-type function now remove both `.gz` and `.xz` suffixes before trying to deduce the file extension. This would not fork for `.gz.xz` files, if we ever need those.
 - No browser actually supports xz, lzma or lzma2 encoding, so the content-encoding part is pretty much useless in this case. Might be better to just set content-type to xz.

One place where I noticed content-type and content-encoding make a difference is when manually downloading files from S3 web interface. In this case `.gz` files get automatically decompressed by the browser and the downloaded file is actually the text file, but with `.gz` extension. This is half-convenient, half-confusing. And the AWS CLI ignores these fields entirely, I think. Not sure we even need this whole content type and encoding business.

Reply:
Setting Content-Type and Content-Encoding (but see caveat below) is a bit friendlier, though maybe not strictly necessary for any of our own code using these files. We do intend these files for the "open" data to be publicly consumed, so friendliness is important to consider there.

Agreed that serially removing the extensions is no good (even though I don't think `.gz.xz` would ever be a concern). See my comment below for a better alternative.

You're right that `Content-Encoding: xz` won't work because clients won't understand it, and this will generally be a fatal error. We should avoid that! Instead, returning `Content-Type: application/x-xz` makes sense I think, though it will mask the contained data type.

You're also right that the Content-Encoding + extension is a bit weird, but it makes it possible to transparently decompress the response, e.g. `curl --compressed … > output` will Just Work, as will fetches from many HTTP client libraries.

Reply:
I meant to add that for downloading, I think we actually want to move away from `aws s3` API operations and instead access via our CDN'd HTTP endpoints (e.g. data.nextstrain.org), which is why HTTP headers matter a bit.